silhouette-password-bcrypt/src/test/scala/com/mohiva/play/silhouette/password/BCryptPasswordHasherSpec.scala

```diff
@@ -17,56 +17,82 @@ package com.mohiva.play.silhouette.password
 
 import com.mohiva.play.silhouette.api.util.PasswordInfo
 import org.specs2.mutable.Specification
+import org.specs2.specification.Scope
 
 /**
  * Test case for the [[BCryptPasswordHasher]] class.
  */
 class BCryptPasswordHasherSpec extends Specification {
 
   "The `hash` method" should {
-    "hash a password" in {
-      val password = "my_S3cr3t_p@sswQrd"
-      val hasher = new BCryptPasswordHasher
-      val info = hasher.hash(password)
-
-      info must beAnInstanceOf[PasswordInfo]
-      info.hasher must be equalTo BCryptPasswordHasher.ID
-      info.password must not be equalTo(password)
-      info.salt must beNone
+    "hash a password" in new Context {
+      override val passwordInfo = hasher.hash(password)
+
+      passwordInfo must beAnInstanceOf[PasswordInfo]
+      passwordInfo.hasher must be equalTo BCryptPasswordHasher.ID
+      passwordInfo.password must not be equalTo(password)
+      passwordInfo.salt must beNone
     }
   }
 
   "The `matches` method" should {
-    "return true if a password matches a previous hashed password" in {
-      val password = "my_S3cr3t_p@sswQrd"
-      val hasher = new BCryptPasswordHasher
-      val info = hasher.hash(password)
+    "return true if a password matches a previous hashed password" in new Context {
+      hasher.matches(passwordInfo, password) must beTrue
+    }
+
+    "return false if a password doesn't match a previous hashed password" in new Context {
+      hasher.matches(passwordInfo, "not-equal") must beFalse
+    }
+  }
 
-      hasher.matches(info, password) must beTrue
```

Comment:
It's possible for both the `hash` and the `matches` method to change in such a way that they are still compatible, but `matches` is no longer compatible with previous hashes. Usually this kind of problem would be solved by comparing the output of `hash` with a hardcoded value, but since it's random we have to resort to double checking.

I suggest `val passwordInfos = Seq(hardcodedPasswordInfo, hasher.hash(password))` in the context and `for (passwordInfo <- passwordInfos)` in the tests.

Taking that further, it's also possible for the tests to pass by luck sometimes even if the methods are broken, and we can reduce that possibility by letting `passwordInfos` contain 5 hardcoded and 5 fresh values.